src/server/mod.rs

```diff
@@ -926,4 +380,572 @@ where
             });
         tokio::spawn(task);
     }
+
+    fn handle_with_auth(session: Arc<Mutex<Session<S, U>>>, next: impl Fn(Event) -> Result<Reply, FTPError>) -> impl Fn(Event) -> Result<Reply, FTPError> {
+        move |event| match event {
+            // internal messages and the below commands are except from auth checks.
+            Event::InternalMsg(_)
```

Comment:
As long as we're cleaning up stuff, can we alphabetize this list? :innocent:

Reply:
he he, yeah of course. I'll make another commit.

Reply:
Will merge this first.